### PR TITLE
ZombieEntity Mappings

### PR DESCRIPTION
**Pull request by beepster4096**
Created a lot of mappings for ZombieEntity. Only unmapped variable is field_7427, as it doesn't appear to do anything.

### CODE REVIEW

File: mappings/net/minecraft/entity/mob/ZombieEntity.mapping [hunk 1]
```diff
@@ -3,6 +3,12 @@ CLASS net/minecraft/class_1642 net/minecraft/entity/mob/ZombieEntity
 		METHOD <init> (Lnet/minecraft/class_1642;Lnet/minecraft/class_1314;DI)V
 			ARG 2 mob
 			ARG 3 speed
+			ARG 5 maxYDifference
+	CLASS class_1644 Data
+		FIELD field_7439 isBaby Z
```
Comment (liach) [inline mappings/net/minecraft/entity/mob/ZombieEntity.mapping]: `baby`. `is` prefix is reserved for methods.
```suggestion
		FIELD field_7439 baby Z
```

File: mappings/net/minecraft/entity/mob/ZombieEntity.mapping [hunk 2]
```diff
@@ -3,6 +3,12 @@ CLASS net/minecraft/class_1642 net/minecraft/entity/mob/ZombieEntity
...
+		METHOD <init> (Lnet/minecraft/class_1642;Z)V
+			ARG 2 isBaby
```
Comment (liach) [inline mappings/net/minecraft/entity/mob/ZombieEntity.mapping]:
```suggestion
			ARG 2 baby
```

File: mappings/net/minecraft/entity/mob/ZombieEntity.mapping [hunk 3]
```diff
@@ -14,18 +20,26 @@ CLASS net/minecraft/class_1642 net/minecraft/entity/mob/ZombieEntity
...
 	METHOD method_7207 getStepSound ()Lnet/minecraft/class_3414;
 	METHOD method_7208 initCustomGoals ()V
 	METHOD method_7209 canConvertInWater ()Z
 	METHOD method_7211 canBreakDoors ()Z
 	METHOD method_7212 shouldBreakDoors ()Z
 	METHOD method_7213 setTicksUntilWaterConversion (I)V
+		ARG 1 ticksUntilWaterConversion
 	METHOD method_7215 getSkull ()Lnet/minecraft/class_1799;
 	METHOD method_7216 burnsInDaylight ()Z
 	METHOD method_7217 setChild (Z)V
```
Comment (liach) [inline mappings/net/minecraft/entity/mob/ZombieEntity.mapping]: consider renaming this to `setBaby` for consistency.

File: mappings/net/minecraft/entity/mob/ZombieEntity.mapping [hunk 4]
```diff
@@ -14,18 +20,26 @@ CLASS net/minecraft/class_1642 net/minecraft/entity/mob/ZombieEntity
 	FIELD field_7434 BABY Lnet/minecraft/class_2940;
 	METHOD <init> (Lnet/minecraft/class_1299;Lnet/minecraft/class_1937;)V
 		ARG 1 type
+	METHOD <init> (Lnet/minecraft/class_1937;)V
+		ARG 1 world
+	METHOD method_20009 (Lnet/minecraft/class_1267;)Z
```
Comment (liach) [inline mappings/net/minecraft/entity/mob/ZombieEntity.mapping:25]: Name this method.

File: mappings/net/minecraft/entity/mob/ZombieEntity.mapping [hunk 5]
```diff
@@ -3,6 +3,12 @@ CLASS net/minecraft/class_1642 net/minecraft/entity/mob/ZombieEntity
 		METHOD <init> (Lnet/minecraft/class_1642;Lnet/minecraft/class_1314;DI)V
 			ARG 2 mob
 			ARG 3 speed
+			ARG 5 maxYDifference
+	CLASS class_1644 Data
+		FIELD field_7439 isBaby Z
+		METHOD <init> (Lnet/minecraft/class_1642;Z)V
+			ARG 2 isBaby
+	FIELD field_19015 difficultyChecker Ljava/util/function/Predicate;
```
Comment (liach) [inline mappings/net/minecraft/entity/mob/ZombieEntity.mapping]: 1. use constant case like `BREAK_DOOR_DIFFICULTY_CHECKER` for static final fields
2. consider mentioning "break door" as this is the only usage of this thing

Comment (beepster4096) [reply mappings/net/minecraft/entity/mob/ZombieEntity.mapping]: Fixed this in 3d5d02f52f449db36f82e4f35a9c5eb5c3689ca5

**Review by liach — APPROVED** [review]
Nice, though I don't know why that 20000 something method cannot be renamed

File: mappings/net/minecraft/entity/mob/ZombieEntity.mapping (hunk 4 above)
Comment (beepster4096) [reply mappings/net/minecraft/entity/mob/ZombieEntity.mapping:25]: This method doesn't show up in Enigma, and there's no method with its signature (Difficulty difficulty).

File: mappings/net/minecraft/entity/mob/ZombieEntity.mapping (hunk 3 above)
Comment (beepster4096) [reply mappings/net/minecraft/entity/mob/ZombieEntity.mapping]: Changed in 13a7075a7d238bc845b639fb7f3473e8d1960441

File: mappings/net/minecraft/entity/mob/MobEntity.mapping [hunk 6]
```diff
@@ -66,6 +66,7 @@ CLASS net/minecraft/class_1308 net/minecraft/entity/mob/MobEntity
 	METHOD method_5938 setSidewaysSpeed (F)V
 		ARG 1 sidewaysMovement
 	METHOD method_5939 canPickupItem (Lnet/minecraft/class_1799;)Z
+		ARG 1 itemStack
```
Comment (Juuxel) [inline mappings/net/minecraft/entity/mob/MobEntity.mapping]: Just `stack` would be enough here.

Comment (beepster4096) [reply mappings/net/minecraft/entity/mob/MobEntity.mapping]: fixed in a3d3b65

**Review by YanisBft — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]
Thanks for this.

**Pull request merged**